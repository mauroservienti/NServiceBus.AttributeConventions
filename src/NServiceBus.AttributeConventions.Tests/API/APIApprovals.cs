using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using PublicApiGenerator;
using System.Runtime.CompilerServices;

namespace NServiceBus.AttributeConventions.Tests.API
{
    public class APIApprovals
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Approve_API()
        {
            var publicApi = typeof(EndpointConfigurationAttributeConventionsExtensions).Assembly.GeneratePublicApi(new ApiGeneratorOptions()
            {
                ExcludeAttributes =
                [
                    "System.Runtime.Versioning.TargetFrameworkAttribute"
                ]
            });

            Approvals.Verify(publicApi.Replace(".git", ""));
        }
    }
}
