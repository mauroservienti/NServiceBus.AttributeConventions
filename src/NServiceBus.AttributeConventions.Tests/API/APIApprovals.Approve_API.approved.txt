[assembly: System.Reflection.AssemblyMetadata("RepositoryUrl", "https://github.com/mauroservienti/NServiceBus.AttributeConventions")]
namespace NServiceBus
{
    public static class EndpointConfigurationAttributeConventionsExtensions
    {
        public static void UseAttributeConventions(this NServiceBus.EndpointConfiguration endpointConfiguration) { }
    }
}